Lib9c/Model/State/StakeState.cs

```diff
@@ -154,7 +155,27 @@ public int CalculateAccumulatedRuneRewards(long blockIndex, out int v1Step, out
             return CalculateStep(blockIndex, startedBlockIndex, out v1Step, out v2Step);
         }
 
-        private int CalculateStep(long blockIndex, long startedBlockIndex, out int v1Step,
+        public int CalculateAccumulatedCurrencyRewards(
+            long blockIndex,
+            out int v1Step,
+            out int v2Step)
+        {
+            var targetBlockIndex = Math.Max(
+                Math.Max(StartedBlockIndex, ReceivedBlockIndex),
+                CurrencyAsRewardStartIndex);
+            v1Step = 0;
+            v2Step = (int)Math.DivRem(
+                blockIndex - targetBlockIndex,
+                RewardInterval,
+                out _
+            );
+            return v1Step + v2Step;
```

Comment:
It seems there is no step interval check (like `CalculateStep` does). If the `RewardInterval` is `100` and `CurrencyAsRewardStartIndex` is `0`, it will receive 1 the first time and 2 the second time, 3 the third time, 6 at all 🤔

Reply:
Ah, `ClaimStakeReward` is updating `StakeState.ReceivedBlockIndex`, so it should be fine.😉